### PR TITLE
Cleanup and Added Status functionality

### PR DESCRIPTION
**Pull request by mikeodr**
Updated urg node with a few commits:
1. roslint
2. moved into a class structure to be able to include and run from processes. Includes adding a thread to run the man `grabScan` and publish of the laser scan.
3. Added function to be able to update the laser scanner status using serial commands via Ethernet socket.

### CODE REVIEW

**Comment by mikeodr** [comment]
@mikepurvis @tonybaltovski

**Comment by mikepurvis** [comment]
Looks reasonable to me.

File: msg/LidarStatus.msg [hunk 1]
```diff
@@ -0,0 +1,10 @@
+uint16 status
+
+uint16 NORMAL=0
+uint16 SETTING=1
+uint16 operating_mode
+
+uint16 area_number
+bool error_status
+uint16 error_code
+bool lockout_status
```
Comment (mikepurvis) [inline msg/LidarStatus.msg]: Comments with documentation references for these fields, please.

Also, this message should probably have a header?

File: package.xml [hunk 2]
```diff
@@ -13,28 +13,37 @@
...
-  <run_depend>urg_c</run_depend>
-  <run_depend>nodelet</run_depend>
-  <run_depend>dynamic_reconfigure</run_depend>
+  <build_depend>std_msgs</build_depend>
   <run_depend>tf</run_depend>
-  <run_depend>laser_proc</run_depend>
-  <run_depend>diagnostic_updater</run_depend>
+  <run_depend>urg_c</run_depend>
+
+  <test_depend>roslint</test_depend>
```
Comment (mikepurvis) [inline package.xml]: You'll need to switch to package format 2 for this to do the thing that you expect (don't even ask).

**Comment by tonybaltovski** [comment]
LGTM :+1:

**Comment by mikeodr** [comment]
Found a bug in some further testing, don't merge yet.

**Comment by mikeodr** [comment]
@tonybaltovski @mikepurvis new updates around thread interaction and reading in the correct amount of data.

File: msg/LidarStatus.msg (hunk 1 above)
Comment (mikeodr) [reply msg/LidarStatus.msg]: Added.

File: package.xml (hunk 2 above)
Comment (mikeodr) [reply package.xml]: Updated.

**Comment by mikeodr** [comment]
"minor" further update making the urg_node logic a library that's link-able by other nodes and imported to provide the ability to do something like this:

```C++
  // Create the laser scanner node
  ros::NodeHandle lnh;
  ros::NodeHandle laser_nh("~/laser");
  urg_node::UrgNode urg(lnh, laser_nh);
  urg.run();
```

File: msg/LidarStatus.msg [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+uint16 SETTING=1
+uint16 operating_mode
+
+# AKA field set
+uint16 area_number
+# If the lidar is reporting an error or not.
+bool error_status
+# THe error code the lidar is reporting.
+uint16 error_code
+# Does the lidar report that it is locked out.
```
Comment (mikepurvis) [inline msg/LidarStatus.msg]: Per recent discussions with SICK, it sounds like they strongly prefer "laser" or "scanner" to "LIDAR". If you don't want to rename the message, please at least capitalize it correctly in these comments.

File: msg/LidarStatus.msg [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+# Normal vs setting in the UAM manual.
+uint16 NORMAL=0
+uint16 SETTING=1
+uint16 operating_mode
+
+# AKA field set
+uint16 area_number
+# If the lidar is reporting an error or not.
+bool error_status
+# THe error code the lidar is reporting.
```
Comment (mikepurvis) [inline msg/LidarStatus.msg]: `THe`

File: msg/LidarStatus.msg (hunk 3 above)
Comment (mikepurvis) [reply msg/LidarStatus.msg]: Given that it's in the `urg_node` package, this message could easily just be `Status`.

File: package.xml [hunk 5]
```diff
@@ -13,28 +13,26 @@
...
-  <run_depend>roscpp</run_depend>
-  <run_depend>rosconsole</run_depend>
-  <run_depend>sensor_msgs</run_depend>
-  <run_depend>urg_c</run_depend>
-  <run_depend>nodelet</run_depend>
-  <run_depend>dynamic_reconfigure</run_depend>
-  <run_depend>tf</run_depend>
-  <run_depend>laser_proc</run_depend>
-  <run_depend>diagnostic_updater</run_depend>
+  <depend>diagnostic_msgs</depend>
```
Comment (mikepurvis) [inline package.xml]: You don't need this dependency explicitly— `diagnostic_updater` will always have it for you.

**Comment by mikepurvis** [comment]
Overall looks great, just nitpicks. 👍

File: msg/LidarStatus.msg (hunk 3 above)
Comment (mikeodr) [reply msg/LidarStatus.msg]: Fixed both.

File: msg/LidarStatus.msg (hunk 4 above)
Comment (mikeodr) [reply msg/LidarStatus.msg]: Fixed.

File: package.xml (hunk 5 above)
Comment (mikeodr) [reply package.xml]: Removed.

**Comment by mikeodr** [comment]
Nit's resolved.

**Comment by mikeodr** [comment]
Tested install space, caught a few missed items there, resolved in last push.
Ready unless there's other comments.

**Comment by mikepurvis** [comment]
LGTM, thanks!

**Pull request merged**